src/System.Net.Sockets/src/System/Net/Sockets/IPPacketInformation.cs

```diff
@@ -49,13 +50,13 @@ public override bool Equals(object comparand)
                 return false;
             }
 
-            IPPacketInformation obj = (IPPacketInformation)comparand;
-            return _address.Equals(obj._address) && _networkInterface == obj._networkInterface;
+            return this == (IPPacketInformation)comparand;
         }
 
         public override int GetHashCode()
         {
-            return _address.GetHashCode() + _networkInterface.GetHashCode();
```

Comment:
If the tests have different results, for support purposes, please add a comment explaining how this is behaving different from desktop.